packages/react-devtools-shared/src/devtools/views/Components/TreeContext.js

```diff
@@ -372,6 +380,92 @@ function reduceTreeState(store: Store, state: State, action: Action): State {
+        if (store.errorsAndWarnings.size === 0) {
+          return state;
+        }
+
+        // Mapping of ID -> Index in the flat tree
+        const elementIndicesWithErrorsOrWarnings: [
+          number,
+          number,
+        ][] = Array.from(store.errorsAndWarnings.keys())
+          .map(elementId => {
+            // $FlowFixMe https://github.com/facebook/flow/issues/1414
+            return [elementId, store.getIndexOfElementID(elementId)];
+          })
+          .filter(([, elementIndex]) => {
+            return elementIndex !== null;
+          });
+        elementIndicesWithErrorsOrWarnings.sort(([, a], [, b]) => {
+          return b - a;
+        });
+
+        const flatSelectedElementIndex =
+          // Selected element in a collapsed tree?
+          selectedElementIndex === null && selectedElementID !== null
+            ? store.getIndexOfElementID(selectedElementID)
+            : selectedElementIndex;
+        const predecessors = elementIndicesWithErrorsOrWarnings.filter(
+          ([, elementIndex]) => {
+            return (
+              flatSelectedElementIndex === null ||
+              elementIndex < flatSelectedElementIndex
+            );
+          },
+        );
+        if (predecessors.length === 0) {
+          selectedElementID = elementIndicesWithErrorsOrWarnings[0][0];
+        } else {
+          selectedElementID = predecessors[0][0];
+        }
+        selectedElementIndex = store.getIndexOfElementID(selectedElementID);
+        lookupIDForIndex = false;
+        break;
+      }
+      case 'SELECT_NEXT_ELEMENT_WITH_ERROR_OR_WARNING_IN_TREE': {
+        if (store.errorsAndWarnings.size === 0) {
+          return state;
+        }
+
+        // Mapping of ID -> Index in the flat tree
+        const elementIndicesWithErrorsOrWarnings: [
+          number,
+          number,
+        ][] = Array.from(store.errorsAndWarnings.keys())
+          .map(elementId => {
+            // $FlowFixMe https://github.com/facebook/flow/issues/1414
+            return [elementId, store.getIndexOfElementID(elementId)];
+          })
+          .filter(([, elementIndex]) => {
+            return elementIndex !== null;
+          });
+        elementIndicesWithErrorsOrWarnings.sort(([, a], [, b]) => {
+          return a - b;
+        });
+
+        const flatSelectedElementIndex =
+          // Selected element in a collapsed tree?
+          selectedElementIndex === null && selectedElementID !== null
+            ? store.getIndexOfElementID(selectedElementID)
+            : selectedElementIndex;
+        const successors = elementIndicesWithErrorsOrWarnings.filter(
+          ([, elementIndex]) => {
+            return (
+              flatSelectedElementIndex === null ||
+              elementIndex > flatSelectedElementIndex
+            );
+          },
+        );
+        if (successors.length === 0) {
+          selectedElementID = elementIndicesWithErrorsOrWarnings[0][0];
```

Comment:
Broke this by loading the test shell with host components hidden, and then unfiltering them and clicking the up arrow.

Reply:
Same error can be triggered by loading with host components visible, hiding them, and then clicking next/prev warning button.